vite.config.ts

```diff
@@ -26,7 +26,7 @@ export default defineConfig({
       fileName: 'index',
     },
     rollupOptions: {
-      external: ['react', 'react-dom'],
+      external: ['react', 'react-dom', 'react/jsx-runtime'],
       output: {
         globals: {
           react: 'React',
```

Comment:
To use `umd` bundles with external imports(`react/jsx-runtime`), then `output.globals` should specify [`id: variableName` pairs necessary for external imports](https://rollupjs.org/configuration-options/#output-globals).

So could you add below?

```diff
globals: {
  react: 'React',
  'react-dom': 'ReactDOM',
+  'react/jsx-runtime': 'react/jsx-runtime'
},
```

then the build should be successful!

Reply:
But the `/` can not be in a var name, right? as far as I understand it needs to be imported into a global variable with the UMD method, like the aliases above are also different from their package name.

Reply:
Oops so

```diff
build: {
+  minify: false,
   ...
}
```

and then 

```diff
- external: ['react', 'react-dom'],
+ external: ['react', 'react-dom', 'react/jsx-runtime'],
   output: {
    globals: {
      react: 'React',
      'react-dom': 'ReactDOM',
    },
    assetFileNames: 'styles/[name].[ext]',
}
```

and then `npm run build` you can check

```javascript
return /* @__PURE__ */ jsxRuntime.jsx("thead", { children: /* @__PURE__ */ jsxRuntime.jsxs("tr", { children: [
      /* @__PURE__ */ jsxRuntime.jsx(Label, { className: "day-of-the-week", textColor, style: { fontSize: cy }, colSpan: 1, children: " " }),
      months.map((month, index2) => {
        let colSpan = colSpans[index2];
```

but still the logs look like this:

```
[vite:dts] Start generate declaration files...
dist/styles/style.css   2.57 kB │ gzip: 0.98 kB
dist/index.js          25.70 kB │ gzip: 6.38 kB
[vite:dts] Declaration files built in 873ms.

No name was provided for external module "react/jsx-runtime" in "output.globals" – guessing "jsxRuntime".
dist/index.umd.cjs  28.74 kB │ gzip: 6.69 kB
```

Sooo,... as it's been used as `jsxRuntime`, how about setting it like below?

```diff
external: ['react', 'react-dom', 'react/jsx-runtime'],
   output: {
    globals: {
        react: 'React',
        'react-dom': 'ReactDOM',
+      'react/jsx-runtime': 'jsxRuntime',
    },
    assetFileNames: 'styles/[name].[ext]',
}
```